consensus/polybft/consensus_runtime.go

```diff
@@ -1085,34 +1091,36 @@ func (c *consensusRuntime) ID() []byte {
 	return c.config.Key.Address().Bytes()
 }
 
-// MaximumFaultyNodes returns the maximum number of faulty nodes based on the validator set
-func (c *consensusRuntime) MaximumFaultyNodes() uint64 {
-	return uint64(c.fsm.validators.Len()-1) / 3
-}
-
-// Quorum returns what is the quorum size for the specified block height
-func (c *consensusRuntime) Quorum(_ uint64) uint64 {
-	return uint64(getQuorumSize(c.fsm.validators.Len()))
-}
-
 // HasQuorum returns true if quorum is reached for the given blockNumber
 func (c *consensusRuntime) HasQuorum(
 	blockNumber uint64,
 	messages []*proto.Message,
 	msgType proto.MessageType,
 ) bool {
-	quorum := c.Quorum(blockNumber)
+	// if we are not using consensus engine for current block return false
+	if c.fsm.parent.Number+1 != blockNumber {
+		c.logger.Warn("HasQuorum checking against stale block",
+			"blockNumber", blockNumber, "parentBlock", c.fsm.parent.Number)
+
+		return false
+	}
 
+	// extract the addresses of all the senders of the messages
+	senders := make([]types.Address, len(messages))
+
+	for i, message := range messages {
+		senders[i] = types.BytesToAddress(message.From)
+	}
+
+	// check quorum
 	switch msgType {
-	case proto.MessageType_PREPREPARE:
-		return len(messages) >= 0
+	default: // proto.MessageType_PREPREPARE etc
```

Comment:
?

Reply:
This is reverted through
https://github.com/0xPolygon/polygon-edge/pull/904/commits/5c5c4689626cc81ca01289e9892811de9eed72be#diff-e5b2e93010ff84c3777aef136e0834d710dd22861ca62ffbab9d5baa12a52b3cR1117-R1118

Reply:
@vcastellm will give explanation